### PR TITLE
Synchronize ByRow and Tables.subset with Tables.jl 1.8

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/3127

### CODE REVIEW

File: src/other/tables.jl [hunk 1]
```diff
@@ -103,3 +103,12 @@ IteratorInterfaceExtensions.getiterator(df::AbstractDataFrame) =
     Tables.datavaluerows(Tables.columntable(df))
 IteratorInterfaceExtensions.isiterable(x::AbstractDataFrame) = true
 TableTraits.isiterabletable(x::AbstractDataFrame) = true
+
+function Tables.subset(df::AbstractDataFrame, inds; view::Union{Bool, Nothing}=nothing)
+    res = view === true ? DataFrames.view(df, inds, :) : df[inds, :]
```
Comment (quinnj) [inline src/other/tables.jl:108]:
```suggestion
    res = view === true ? view(df, inds, :) : df[inds, :]
```
don't need the `DataFrames.` qualifier, right?

Comment (bkamins) [reply src/other/tables.jl:108]: you need it as `view` is a kwarg name in this function

File: Project.toml [hunk 2]
```diff
@@ -36,7 +36,7 @@ Reexport = "0.1, 0.2, 1"
 ShiftedArrays = "1"
 SortingAlgorithms = "0.1, 0.2, 0.3, 1"
 TableTraits = "0.4, 1"
-Tables = "1.2"
+Tables = "1.8"
```
Comment (bkamins) [inline Project.toml:39]:
```suggestion
Tables = "1.8.1"
```

**Comment by bkamins** [comment]
@nalimilan - could you please review this. Due to changes in Tables.jl 1.8 tests in DataFrames.jl will fail in all other PRs, so this PR needs to be merged first and next all other PRs must be updated against new `main`.

File: src/other/tables.jl [hunk 3]
```diff
@@ -103,3 +103,12 @@ IteratorInterfaceExtensions.getiterator(df::AbstractDataFrame) =
     Tables.datavaluerows(Tables.columntable(df))
 IteratorInterfaceExtensions.isiterable(x::AbstractDataFrame) = true
 TableTraits.isiterabletable(x::AbstractDataFrame) = true
+
+function Tables.subset(df::AbstractDataFrame, inds; view::Union{Bool, Nothing}=nothing)
```
Comment (bkamins) [inline src/other/tables.jl:107]:
```suggestion
@inline function Tables.subset(df::AbstractDataFrame, inds; view::Union{Bool, Nothing}=nothing)
```

[the remark this answers is not in the document]
Comment (bkamins) [reply src/other/tables.jl]: added. It should not be a problem.

**Review by nalimilan — APPROVED** [review]

File: src/other/tables.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (bkamins) [reply src/other/tables.jl]: I added `@inferred` where applicable

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**